pilot/pkg/model/credentials/resource.go

```diff
@@ -61,7 +61,7 @@ type SecretResource struct {
 }
 
 func (sr SecretResource) Key() string {
-	return sr.ResourceType + "/" + sr.ResourceKind.String() + "/" + sr.Name + "/" + sr.Namespace + "/" + string(sr.Cluster)
+	return sr.ResourceName + "/" + sr.ResourceType + "/" + sr.ResourceKind.String() + "/" + sr.Name + "/" + sr.Namespace + "/" + string(sr.Cluster)
```

Comment:
how could it fix the original issue.

Reply:
- Without this fix, it will hit the cache and return the secret with wrong resource name (without namespace)
- With this fix, there will be two entries in the cache , one with namespace and one without. When Envoy sends the new request, we send the correct cached secret

Reply:
@hzxuzhonghu do you feel strongly about changing the resource name to include namespace also if not provided? - wasn't sure what else can break with that

Reply:
I think I agree with @hzxuzhonghu here; it feels cleaner for the xDS resource name to be normalized to the absolute location (so to speak) of the secret

Reply:
Hmm ok in that case, I'm in favor of stability. Sorry for the back in forth, but GREAT call out re:upgrades. I think it only affects in place control plane upgrades but could still be disruptive to users for little gain

Reply:
Ok. No worries I will revert back and just have previous logic